### PR TITLE
Feature/30256 test card

### PR DESCRIPTION
**Pull request by SandipMehta**
Card component unit tests.

### CODE REVIEW

File: src/components/Card/tests/Card.test.tsx [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+                                        <p>View a summary of your online store’s performance.</p>
+                                   </Card>);
+        expect(cardWrapper.prop('sectioned')).toBe(true);
+    });
+    it('Verify sectioned when it is not set', () => {
+        const cardWrapper = mount( <Card title="Online store dashboard - Card" subdued>
+                                        <p>View a summary of your online store’s performance.</p>
+                                   </Card>);
+        expect(cardWrapper.prop('sectioned')).toBeFalsy();
+        cardWrapper.setProps({sectioned: false});
```
Comment (mkulinski) [inline src/components/Card/tests/Card.test.tsx]: `{sectioned: false}` should be `{ sectioned: false }`

File: src/components/Card/tests/Card.test.tsx [hunk 2]
```diff
@@ -0,0 +1,136 @@
+import * as React from 'react';
+import { mount } from 'enzyme';
+import Card from '../Card';
+import Section from '../Section';
+import {Action} from '../../../types';
```
Comment (mkulinski) [inline src/components/Card/tests/Card.test.tsx]: `{Action}` should be `{ Action }`

File: src/components/Card/tests/Card.test.tsx [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+
+describe('Card - Test Suit', () => {
+    const theme = {Card: 'Card',
```
Comment (mkulinski) [inline src/components/Card/tests/Card.test.tsx]:
```
{Card: 'Card',
 +    subdued: 'subdued',
 +    Header: 'Header',
 +    Section: 'Section',
 +    SectionHeader: 'SectionHeader',
 +    Footer: 'Footer',
 +    };
```
Should be: 
```
{
  Card: 'Card',
  subdued: 'subdued',
  Header: 'Header',
  Section: 'Section',
  SectionHeader: 'SectionHeader',
  Footer: 'Footer',
};
```

File: src/components/Card/tests/Card.test.tsx [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+        const spy = jest.fn();
+        const action: Action = {content: 'Action Content', onAction: () => {spy(); }};
+        const cardWrapper = mount( <Card title="Online store dashboard - Card"  secondaryFooterAction={action}  >
+                                        <p>View a summary of your online store’s performance.</p> 
+                                   </Card>);
+        cardWrapper.find('button').simulate('click');
+        expect(spy).toHaveBeenCalled();
+    });
+    it('Verify sectioned is boolean type', () => {
+        const cardWrapper = mount( <Card title="Online store dashboard - Card" sectioned >
```
Comment (mkulinski) [inline src/components/Card/tests/Card.test.tsx]: no space between sectioned and >

File: src/components/Card/tests/Card.test.tsx [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+                                        <p>View a summary of your online store’s performance.</p> 
+                                   </Card>);
+        expect(typeof cardWrapper.prop('sectioned')).toBe('boolean');
+    });
+    it('Verify subdued is boolean type', () => {
+        const cardWrapper = mount( <Card title="Online store dashboard - Card" subdued >
```
Comment (mkulinski) [inline src/components/Card/tests/Card.test.tsx]: no space between subdued and >

File: src/components/Card/tests/Card.test.tsx [hunk 6]
```diff
@@ -0,0 +1,136 @@
...
+                                        </Section>
+                                   </Card>);
+        expect(cardWrapper.prop('sectioned')).toBe(true);
+        expect(cardWrapper.find('Section').at(1).text()).toBe('ReportsView a summary of your online store’s performance.');
+        expect(cardWrapper.find('Section').at(2).text()).toBe('Summary ReportsView a summary of your online store’s performance, including sales, visitors, top products, and referrals.');
+        expect(cardWrapper.find('h3').length).toBe(2);
+        expect(cardWrapper.find('p').length).toBe(2);
+    });
+    it('Verify Card with subdued option.', () => {
+        const cardWrapper = mount( <Card title="Online store dashboard - Card" sectioned >
```
Comment (mkulinski) [inline src/components/Card/tests/Card.test.tsx]: no space between sectioned and >

File: src/components/Card/tests/Card.test.tsx [hunk 7]
```diff
@@ -0,0 +1,136 @@
...
+                                            <p>View a summary of your online store’s performance, including sales, visitors, top products, and referrals.</p>
+                                        </Section>
+                                   </Card>);
+        expect(cardWrapper.find('h3').length).toBe(2);
+        expect(cardWrapper.find('p').length).toBe(2);
+        expect(cardWrapper.find('h3').at(0).text()).toBe('Reports');
+        expect(cardWrapper.find('h3').at(1).text()).toBe('Summary Reports');
+    });
+    it('Verify Card with theme.', () => {
+        const cardWrapper = mount( <Card title="Online store dashboard - Card" sectioned >
```
Comment (mkulinski) [inline src/components/Card/tests/Card.test.tsx]: no space between sectioned and >

File: src/components/Card/tests/Card.test.tsx [hunk 8]
```diff
@@ -0,0 +1,136 @@
...
+                                        <Section title="Reports" subdued >
```
Comment (mkulinski) [inline src/components/Card/tests/Card.test.tsx]: no space between subdued and >

**Comment by SandipMehta** [comment]
code review fixes are done.

**Pull request merged**